### PR TITLE
add username generation to generator

### PR DESCRIPTION
**Pull request by kspearrin**
## Type of change

- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Add support for username generation.

Asana task: https://app.asana.com/0/1200578627114106/1201532987437763

## Code changes

- Add new translations
- Implement UI changes to accommodate username generation on the generator page (see screenshots below)

## Screenshots

image

image

image

## Testing requirements

1. Test existing password generation options from generator.
2. Test new username generation options from generator.

## Before you submit

- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [x] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: src/app/tools/password-generator.component.html [hunk 1]
```diff
@@ -1,182 +1,305 @@
...
+          <label for="includeNumberUsername" class="form-check-label">{{
+            "includeNumber" | i18n
+          }}</label>
+        </div>
+      </div>
+    </div>
+  </ng-container>
 </ng-container>
-<div class="d-flex">
+<div class="d-flex" *ngIf="type === 'password'">
```
Comment (eliykat) [inline src/app/tools/password-generator.component.html:302]: You can roll this into the `<ng-container>` for password generation.

File: src/app/tools/password-generator.component.html [hunk 2]
```diff
@@ -196,4 +319,12 @@ <h1>{{ "passwordGenerator" | i18n }}</h1>
     </button>
   </div>
 </div>
+<div *ngIf="type === 'username'">
```
Comment (eliykat) [inline src/app/tools/password-generator.component.html:322]: Same here, move into the `<ng-container>` for username generation.

**Review by eliykat — CHANGES_REQUESTED** [review]

**Review by eliykat — COMMENTED** [review]
In the desktop PR you added a "generate" action button next to the username field in `add-edit-component`. Can we do the same here?

**Comment by kspearrin** [comment]
> In the desktop PR you added a "generate" action button next to the username field in `add-edit-component`. Can we do the same here?

We don't have the concept of opening a dedicated generator view in the web vault. For password we just generate a password based on your last settings. I am not sure that will be sufficient enough for a username generator where you probably need to adjust the settings to the website more frequently.

**Review by eliykat — APPROVED** [review]

**Pull request merged**